### DISCUSSION TITLE
Orca Slicer Mobile App?

### DISCUSSION THREAD

**Discussion by Im2code** [OrcaSlicer/OrcaSlicer#1711] [Ideas]
I've been looking around for mobile slicers, but I can't seem to find any good options. There's AstroPrint with its cloud slicing but it doesn't have all the features I wanted and the UI is not very pleasing. So is it possible for Orca Slicer to have an IOS app for iPads or an Android app?

**Comment by RaySajuuk** [comment]
I would assume maintaining current slicer is already a tremendous work, take a look at the Issues section. The mount of resources required to build a slicer on mobile will be a completely different level.

This will almost not going to happen. Put in an analogy, OrcaSlicer is like modification to a car, there is a lot of work to do but the engine and core components are provide for. What you are asking is to convert a gasoline engine car to an electric car. Or put it simply, this is way to much to ask for.

**Reply by ju5t3nc4s3** [reply]
Then build a android app to remotly control orca on your network. Or build orka into a server

**Reply by RaySajuuk** [reply]
This is an open source project, new features are maintained by people who share the interest. Especially for what you are suggesting is not small project. So if you are merely asking for those features without any contribution to the code base then it is not helpful.

**Reply by S117X** [reply]
It boots

**Comment by CruisnGrrl** [comment]
With the power of the M series processors in the iPads they should be able to handle the slicing requirements. Not sure what would be involved porting the ARM64 macOS release to iOS

**Reply by S117X** [reply]
M5 MacBook Air Using Supergrok heavy atm told it to port it! lets see what it does

**Reply by mazzanfar** [reply]
Any updates one that?

**Comment by vgdh** [comment]
if the mobile device is capable to slice it can do linux VM as well. 😁

**Reply by CruisnGrrl** [reply]
I'm not aware of any linux VMs for iPadOS (iOS), I do know that the iPad Air, iPad Pro and the Apple Silicon Macs all share the same processor and that processor is a beast and the Macs can run iOS apps natively.

**Reply by vgdh** [reply]
Orca already has build for Macs. May it be run on iOS?

**Reply by CruisnGrrl** [reply]
Sadly it would need to be compiled for iOS for release on the App Store, I have seen some programs release on iOS to be used with both iPads and Macs to reach a wider audience.

**Reply by pbaerthel** [reply]
> Sadly it would need to be compiled for iOS for release on the App Store, I have seen some programs release on iOS to be used with both iPads and Macs to reach a wider audience.

The problem is, iOS has AppStore restrictions, they don't allow to upload Open Source Apps to the AppStore. CPU power should be enough (have an M1 MacBook and an M1 iPad Pro, so it should perform almost the same). BUT, the Orca Team could release an iPad App to an alternative App Market (now possible in EU) and this is what I'm hoping for.

**Reply by s-predybailo** [reply]
> they don't allow to upload Open Source Apps to the AppStore

The AppStore allows open-source apps as long as they comply with all platform requirements. Examples include VLC, Signal, and Firefox.

**Reply by relink2013** [reply]
Figured I'd throw in my two cents even though this is a bit old. @vgdh mentioned an idea that is 100% possible, no jailbereak, no side loading and I cant think of any reason it wouldn't work.

> if the mobile device is capable to slice it can do linux VM as well.

> I'm not aware of any linux VMs for iPadOS (iOS)

100% doable, there is an app in the AppStore called "UTM", it's been there for a while now. It's for managing and creating VMs on device. Think of it like VirtualBox, except under the hood it's using a version of QEMU optimized for Apple Silicon. 

UTM can run ARM and x86 OSes perfectly fine. QEMU already supported CPU emulation to begin with but I think a lot of people forgot that Apple designed their chips to be able to efficiently run x86 code for backwards compatibility. They don't have AVX2 or anything like that, but they can execute x86 instructions.

I've ran Windows 98 on my iPhone literally for no reason other than I could, so it doesn't even need to be an M-series CPU.

> Sadly it would need to be compiled for iOS for release on the App Store

Why is that sad? If it can run on Apple Silicon it can run on an iPad. I think the biggest hurdle for an iPad version would be making sure the UI actually works properly with touch only. If Orca Slicer can already run on ARM then porting to iOS isn't as big of a stretch as people probably think it is. 

For example, DaVinci Resolve is in the AppStore for iPad, and not a stripped down version, Full fat DaVinci Resolve! Though they hadn't (last I checked) fully optimized it for an all touch UI. It actually required a keyboard for several features, not sure if this has changed since I last used it. 

> The problem is, iOS has AppStore restrictions, they don't allow to upload Open Source Apps to the AppStore.

That is demonstrably not true, there are tons of FOSS apps in the AppStore, several of which I use on a daily basis. 

Sorry if my comment got a bit long winded, but the idea purposed is more than likely totally possible, and would probably run just fine. It's a legit idea that may actually help someone, and it just got passed over because "Apple Bad"...

This whole idea of "Apple won't let you" is ancient, nearly all of the "restrictions" I see people bring up the most have long since been removed, some as long ago as iOS 7...that was over 12 years ago btw.

**Reply by S117X** [reply]
I hope grok knows about this lol

**Comment by CruisnGrrl** [comment]
But there are open source apps on iOS. Here's an example https://apps.apple.com/ca/app/opendocument-reader-view-odt/id1510195065LisaOn Jun 20, 2024, at 14:28, pbaerthel ***@***.***> wrote:﻿

Sadly it would need to be compiled for iOS for release on the App Store, I have seen some programs release on iOS to be used with both iPads and Macs to reach a wider audience.

The problem is, iOS has AppStore restrictions, they don't allow to upload Open Source Apps to the AppStore. CPU power should be enough (have an M1 MacBook and an M1 iPad Pro, so it should perform almost the same). BUT, the Orca Team could release an iPad App to an alternative App Market (now possible in EU) and this is what I'm hoping for.

**Comment by pbaerthel** [comment]
Yes but now in EU, Sideloading is allowed, so it could be possible


Am 06.07.2024 um 01:36 schrieb s-predybailo ***@***.***>:

﻿

they don't allow to upload Open Source Apps to the AppStore

The AppStore allows open-source apps as long as they comply with all platform requirements. Examples include VLC, Signal, and Firefox.

**Reply by S117X** [reply]
then we will comply!

**Comment by S117X** [comment]
its 2026 lets see is Super Grok Heavy can do it, currently downloading iOS 26 Emulator to start porting. If not I will Use Fable.

**Reply by S117X** [reply]
Grok still working

**Comment by S117X** [comment]
<img width="401" height="755" alt="image" src="https://github.com/user-attachments/assets/e20e05f0-b85d-42e8-b2fe-38b901ca4eaa" />

**Comment by S117X** [comment]
https://github.com/user-attachments/assets/1ec3bed4-1c00-4c8d-9c7f-000722c23f0e

**Comment by gurmad** [comment]
Th
